lib/msal-common/src/auth/ClientInfo.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ClientAuthError } from "../error/ClientAuthError";
+import { StringUtils } from "../utils/StringUtils";
+import { ICrypto } from "../utils/crypto/ICrypto";
+
+/**
+ * @hidden
+ */
+export class ClientInfo {
+
+    private _uid: string;
+    get uid(): string {
+        return this._uid ? this._uid : "";
+    }
+
+    set uid(uid: string) {
+        this._uid = uid;
+    }
+
+    private _utid: string;
+    get utid(): string {
+        return this._utid ? this._utid : "";
+    }
+
+    set utid(utid: string) {
+        this._utid = utid;
+    }
+
+    constructor(rawClientInfo: string, crypto: ICrypto) {
+        if (!rawClientInfo || StringUtils.isEmpty(rawClientInfo)) {
```

Comment:
are there times there are empty client info and we want to construct the class?

Reply:
I believe there are cases in B2C where we may not receive the client info object, but I will double check to verify. It would probably be a better pattern to throw a warning and not create the object at all, so I will make some changes to reflect that.

Reply:
Updated, please re-review

Reply:
Some cases of device flow had this issue but eSTS guarantees client_info and if it is empty, we need to raise a bug against them. However it may be necessary to confirm the expected behavior from msal js in case it returns empty.